feat(wip): add mobile styling on structure and poll

ynnxj opened the pull request:
A v messy commit history, sorrryyyyy.

Kattenelvis committed:
Merge pull request #209 from Kattenelvis/dev

v40

Kattenelvis committed:
Merge pull request #211 from Kattenelvis/dev

fixed build error

ynnxj committed:
feat(styling): fix mobile timeline

Kattenelvis committed:
Merge pull request #212 from Kattenelvis/dev

v41

Kattenelvis committed:
Merge pull request #213 from Kattenelvis/dev

v41

Kattenelvis committed:
Merge pull request #214 from Kattenelvis/dev

hotfix

ynnxj committed:
fix: remove mobile-support blocker

Temporarily to work on responsive design

Kattenelvis committed:
Merge pull request #215 from Kattenelvis/dev

Hotfix

ynnxj committed:
fix: tweak mobile timeline styling

ynnxj committed:
feat: responsive poll layout

ynnxj committed:
refactor: replace comments with descriptive class names

ynnxj committed:
fix: mobile proposal button positioning

ynnxj committed:
fix: style and move button for timeline info

ynnxj committed:
fix: mobile timeline on feed

ynnxj committed:
feat: make Structure mobile-responsive

Add mobile detection.
Simplify Timeline with props.
Add flex column to mobile.

ynnxj committed:
feat: mobile-responsive poll phases 1-3

Combine both left- and right-slots into single column on mobile.
Add return buttons on mobile.

ynnxj committed:
fix: center pie chart on poll phase 7

ynnxj committed:
Merge branch 'dev' of https://github.com/ynnxj/flowback-frontend into dev

ynnxj committed:
fix: update history component with new structure prop

Show all slots on mobile in a column.

ynnxj committed:
fix: enhance mobile layout flexibility

Add new boolean prop for showing both slots on mobile.
Enhance conditional rendering.

ynnxj committed:
fix: styling on create-prediction component and buttons

Add mobile styling.
Buttons originally overlapped content.

ynnxj committed:
fix: add mobile styling to poll phase 3-7

Add mobile layout props to Structure.
Add mobile-only return buttons.
Update styling.

Kattenelvis commented on src/lib/Delegation/History.svelte:
You don't need to have the "={true}".

ynnxj replied:
Oop-

Kattenelvis commented on src/lib/Poll/NewDesign/Structure.svelte:
```diff
@@ -9,11 +10,25 @@
 		resetScroll = false;
 
+	export let showRightOnMobile = false;
+	export let showBothSlotsOnMobile = false;
```
You don't need to repeat export let

ynnxj replied:
Made it with an array instead.

Kattenelvis commented on src/lib/Poll/NewDesign/Structure.svelte:
```diff
@@ -9,11 +10,25 @@
 
+  onMount(() => {
+    const checkMobile = () => {
+      isMobile = window.innerWidth < 768;
```
Have you considered landscape view

ynnxj replied:
I’ll check it out.

ynnxj replied:
I think this is something I'll be doing eventuelly, not right now. I'll work on it!

Kattenelvis commented on src/lib/Poll/NewDesign/Timeline.svelte:
```diff
@@ -127,7 +127,7 @@
 	{#if enableDetails && displayDetails}
 		<button
-			class="hover:underline flex items-center gap-1 text-xs"
+			class="hover:underline flex items-center gap-1 cursor-pointer text-xs"
```
Why are you re-introducing this?

ynnxj replied:
Can you clarify which part you mean?

Kattenelvis replied:
Well based on code-highlightning there's just one I'm referring to. There shouldn't be a pointer cursor here, since one cannot click to get to a new page or that performs some kind of action

ynnxj replied:
Oh, I didn't see it for some reason. But yeah, no, it just accidentally came back somehow, hehe. I'll remove it.

Kattenelvis commented on src/lib/Poll/NewDesign/Structure.svelte:
```diff
@@ -45,16 +61,43 @@
-			<slot name="left" class="h-full" />
-		</div>
+	{#if isMobile}
+		{#if showBothSlotsOnMobile}
```
You're adding so many conditionals, which makes it hard to maintain and repeats code. Surely there's better ways to do this with say, the :md, :sm etc tailwind tags without requiring this rebulding of Structure? I might be wrong, this might be needed, but have you doubble-checked this?

ynnxj replied:
I agree. I didn’t want to mess with the original structure too much and had a hard time making it work properly. I’ll look into a better solution and get back to you!

ynnxj replied:
I did my best simplifying the structure by reducing the conditionals and moving logic to computed properties. Let me know what you think.

Kattenelvis commented on src/routes/groups/[groupId]/polls/[pollId]/+page.svelte:
```diff
@@ -129,16 +129,16 @@
+			<!-- PHASE 2: PROPOSAL CREATION -->
 			{:else if phase === 'proposal'}
-				<Structure bind:phase bind:poll bind:resetScroll>
+				<Structure bind:phase bind:poll bind:resetScroll showRightOnMobile={!!selectedProposal || displayForm}>
```
Double Negation??

Kattenelvis replied:
I guess it goes from null to true to false, so that sorta makes sense. But JavaScript already type coerces null into false in conditional expressions

ynnxj replied:
I agree, it hurts my brain a bit ngl. Is this change better?

Kattenelvis commented on src/routes/groups/[groupId]/polls/[pollId]/+page.svelte:
```diff
@@ -129,16 +129,16 @@
-				</Structure>
+			<Structure bind:phase bind:poll>
+				<div slot="left" class="h-full"><AreaVote /></div>
+				<div slot="bottom" class="!p-0">
```
I'm gona discuss with Loke if we want this change

Kattenelvis replied:
Actually it doesn't matter much, he wants to get rid of area vote regardless. Having it at the bottom is fine, but you don't need to work on this phase much

Kattenelvis commented on src/routes/groups/[groupId]/polls/[pollId]/+page.svelte:
```diff
@@ -203,19 +213,35 @@
+							Class="w-full mt-auto"
 							buttonStyle="primary-light"
-							disabled={displayForm && !selectedProposal}
+							disabled={proposalsToPredictionMarket.length === 0}
```
I like this, this is a cleaner statement

Kattenelvis commented on src/routes/groups/[groupId]/polls/[pollId]/+page.svelte:
```diff
@@ -336,6 +396,17 @@
+									}}
+										buttonStyle="primary-light"
+									>
+										<Fa icon={faArrowLeft} />
```
You're really repeating this left-arrow thing alot. This might be fine, but I'm wondering if there's a way to not repeat it so much

ynnxj replied:
Should I make a reusable component out of it instead?

ynnxj replied:
Well, I made it into a componant so it looks cleaner now. Let me know what you think.

ynnxj committed:
refactor: simplify structure and mobile logic

Move mobile conditions and grid class to computed props.

Simplify structure template.

ynnxj committed:
refactor: update mobile poll phases

Replace double negation in mobile props with null checks.

Use computed props for mobile display logic for improved readability.

ynnxj committed:
chore: remove cursor-pointer from button

ynnxj committed:
feat: create back button for mobile

Create back button component to keep code DRY.

ynnxj committed:
chore: delete console.log

Kattenelvis committed:
Merge branch 'dev' into 11-fix-responsive-layout-design
